pkg/providers/vsphere/controlplane_test.go

```diff
@@ -125,6 +125,46 @@ func TestControlPlaneSpecNoKubeVersion(t *testing.T) {
 	g.Expect(err).To(MatchError(ContainSubstring("generating vsphere control plane yaml spec")))
 }
 
+func TestControlPlaneSpecUpdateMachineTemplates(t *testing.T) {
+	g := NewWithT(t)
+	logger := test.NewNullLogger()
+	ctx := context.Background()
+	spec := givenClusterSpec(t, testClusterConfigMainFilename)
+	originalKubeadmControlPlane := kubeadmControlPlane()
+	originalEtcdCluster := etcdCluster()
+	originalEtcdCluster.Spec.InfrastructureTemplate.Name = "test-etcd-2"
+	client := test.NewFakeKubeClient(
+		originalKubeadmControlPlane,
+		originalEtcdCluster,
+		vsphereMachineTemplate("test-control-plane-1"),
+		vsphereMachineTemplate("test-etcd-2"),
+	)
+	spec.VSphereMachineConfigs["test-etcd"].Spec.Datastore = "new-datastore"
+
+	kcp := wantKCP()
```

Comment:
```suggestion
	wantKCP := originalKubeadmControlPlane.DeepCopy()
```

you probably want to do this before passing the original ones to the client